docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.install

```diff
@@ -36,3 +53,84 @@ function hs_paragraph_types_update_9401() {
     $paragraph->save();
   }
 }
+
+/**
+ * Update postcard body field text format.
+ */
+function hs_paragraph_types_update_9402() {
+  // Load filter_format and editor exported configuration.
+  $config_directory = new FileStorage(Settings::get('config_sync_directory'));
+  $filter_format_config = $config_directory->read('filter.format.minimal_html_with_styles');
+  $editor_config = $config_directory->read('editor.editor.minimal_html_with_styles');
+  if (!FilterFormat::load('minimal_html_with_styles')) {
+    $filter_format = FilterFormat::create($filter_format_config);
+    $filter_format->save();
+  }
+  if (!Editor::load('minimal_html_with_styles')) {
+    $editor = Editor::create($editor_config);
+    $editor->save();
+  }
+
+  // Update field_hs_postcard_body field defnition exported configuration.
+  $field_config = \Drupal::configFactory()->getEditable('field.field.paragraph.hs_postcard.field_hs_postcard_body');
+  $field_config->set('third_party_settings.allowed_formats.allowed_formats.0', 'minimal_html_with_styles');
+  $field_config->save();
+}
+
+/**
+ * Update user permissions for minimal_html_with_styles text format.
+ */
+function hs_paragraph_types_update_9403() {
+  // Update contributor role permissions.
+  $role_config = \Drupal::configFactory()->getEditable('user.role.contributor');
+  $existing_permissions = $role_config->get('permissions');
+  $existing_permissions[] = 'use text format minimal_html_with_styles';
+  $role_config->set('permissions', $existing_permissions);
+  $role_config->save();
+
+  $role_config = \Drupal::configFactory()->getEditable('user.role.site_manager');
+  $existing_permissions = $role_config->get('permissions');
+  $existing_permissions[] = 'use text format minimal_html_with_styles';
+  $role_config->set('permissions', $existing_permissions);
+  $role_config->save();
+}
+
+/**
+ * Update field format for existing postcard paragraph content.
+ */
+function hs_paragraph_types_update_9404(&$sandbox = NULL) {
+  $storage_handler = \Drupal::entityTypeManager()->getStorage('paragraph');
+  if (!isset($sandbox['total'])) {
+    // Get all existing instances of hs_postcard.
+    $query = $storage_handler->getQuery()
+      ->accessCheck(TRUE)
+      ->condition('type', 'hs_postcard')
+      ->condition('field_hs_postcard_body', NULL, 'IS NOT NULL');
+    $total = $query->count()->execute();
+    $sandbox['total'] = $total;
+    $sandbox['current'] = 0;
+  }
+  $batch = 25;
+
+  $query = $storage_handler->getQuery()
+    ->accessCheck(TRUE)
+    ->condition('type', 'hs_postcard')
+    ->condition('field_hs_postcard_body', NULL, 'IS NOT NULL')
+    ->range($sandbox['current'], $batch);
+  $paragraph_ids = $query->execute();
+
+  $postcard_paragraphs = $storage_handler->loadMultiple($paragraph_ids);
```

Comment:
just store the ids in the sandbox, and do a splice `$ids = array_splice($sandbox['ids'], 0, 25);

Reply:
Fixed